server/rest.go

```diff
@@ -1271,29 +1284,33 @@ func (s *Server) InsertContract(c echo.Context) error {
 		return api.Invalid.Build(c)
 	}
 	ctx := context.Background()
-	if strings.EqualFold(addrInfo.ErcTypes, "KRC20") {
-		smcABIStr, err := s.dbClient.SMCABIByType(ctx, addrInfo.ErcTypes)
-		if err == nil {
-			abiData, err := base64.StdEncoding.DecodeString(smcABIStr)
-			if err != nil {
-				s.logger.Warn("Cannot decode smc abi", zap.Error(err))
-			}
-			jsonABI, err := abi.JSON(bytes.NewReader(abiData))
-			if err != nil {
-				s.logger.Warn("Cannot convert decoded smc abi to JSON abi", zap.Error(err))
-			}
-			if err == nil {
-				totalSupply, _ := s.kaiClient.GetKRCTotalSupply(ctx, &jsonABI, common.HexToAddress(addrInfo.Address))
-				if totalSupply != nil {
-					addrInfo.TotalSupply = totalSupply.String()
-				}
-			}
-		}
+	krcTokenInfoFromRPC, err := s.getKRCTokenInfoFromRPC(ctx, addrInfo.Address, addrInfo.KrcTypes)
+	if err != nil && strings.HasPrefix(addrInfo.KrcTypes, "KRC") {
```

Comment:
Condition looks wrong? maybe !strings.HasPrefix  ? And use this check here 
if krcTokenInfoFromRPC != nil

Reply:
We only get info of KRC tokens (`strings.HasPrefix(addrInfo.KrcTypes, "KRC")`) here. If this contract is KRC and we got error when try to retrieve its info from network then better return `Invalid` before making any changes to db/cache.